RequestService/RequestService.Handlers/GetAllJobsByFilterHandler.cs

```
@@ -0,0 +1,204 @@
+                {
+                    LocationsRequests = new HelpMyStreet.Contracts.AddressService.Request.LocationsRequest()
+                    {
+                        Locations = distinctLocations.Select(x => x.Value).ToList()
+                    }
+                };
+                var details = await _addressService.GetLocations(locationRequest);
+
+                if (details != null)
+                {
+                    locationDetails = details.LocationDetails;
+                }
+            }
+
+            if (locationDetails.Count > 0)
+            {
+                foreach (JobDTO j in allJobs.Where(x => x.Location != null))
+                {
+                    var location = locationDetails.First(x => x.Location == j.Location);
+
+                    if (location != null)
+                    {
+                        j.PostCode = location.Address.Postcode;
+                    }
+                }
+            }
+        }
+
+        public async Task<GetAllJobsByFilterResponse> Handle(GetAllJobsByFilterRequest request, CancellationToken cancellationToken)
+        {
+            if (!string.IsNullOrEmpty(request.Postcode))
+            {
+                request.Postcode = HelpMyStreet.Utils.Utils.PostcodeFormatter.FormatPostcode(request.Postcode);
+
+                try
+                {
+                    var postcodeValid = await _addressService.IsValidPostcode(request.Postcode, cancellationToken);
+                }
+                catch (HttpRequestException)
+                {
+                    throw new PostCodeException();
+                }
+            }
+
+            if (request.ExcludeSiblingsOfJobsAllocatedToUserID.HasValue)
+            {
+                if (request.JobStatuses?.JobStatuses.Count() == 0 || !request.JobStatuses.JobStatuses.Contains(JobStatuses.Open))
+                {
+                    throw new InvalidFilterException();
```

Should we also throw an InvalidFilterException if an AllocatedToUserId is supplied along with a JobStatus which is not Accepted/InProgress/Complete?